front/src/modules/trainschedule/components/Timetable/TimetableToolbar.tsx

```diff
@@ -117,11 +118,16 @@ const TimetableToolbar = ({
     }
   };
 
-  const handleTrainsDelete = async () => {
+  const handleTrainsDelete = async (currentSelectedTrainId?: TrainId) => {
     const itemsCount = selectedTimetableItemIds.length;
 
-    // TODO Paced train : Adapt this to handle paced trains selection in issue https://github.com/OpenRailAssociation/osrd/issues/11054
-    if (selectedTrainId && selectedTimetableItemIds.includes(selectedTrainId as TrainScheduleId)) {
+    const isSelectedTimetableItemInSelection = selectedTimetableItemIds.some((timetableItemId) =>
```

Comment:
Before using `some`, you can test if `selectedTrainId` is not null
```suggestion
    const isSelectedTimetableItemInSelection = selectedTrainId && selectedTimetableItemIds.some((timetableItemId) =>
```

Reply:
Done